firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -900,6 +899,47 @@ private ListenerRegistration addSnapshotListenerInternal(
         firestore.getClient(), queryListener, activity, wrappedListener);
   }
 
+  @PublicApi
+  public String getPath() {
+    return query.getPath().canonicalString();
+  }
+
+  @PublicApi
+  public List<com.google.firebase.firestore.Filter> getFilters() {
```

Comment:
Note to self: Since we're exposing filters as `Filter` objects, should we have a `Query.addFilter(...)` method in order to re-construct the query?  Else, you'd need to do a `switch` on the operator and then call the appropriate `Query.where...()` method to match the operator which would be tedious.